postgres/postgres.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2019 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package postgres provides functions to open PostgreSQL databases with OpenCensus instrumentation.
+
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net/url"
+)
+
+type URLOpener struct{}
+
+// Scheme is the URL scheme postgres registers its URLOpener under on
+// DefaultMux.
+const Scheme = "postgres"
+
+func init() {
+	DefaultURLMux().RegisterPostgres(Scheme, &URLOpener{})
+}
+
+func (*URLOpener) OpenPostgresURL(ctx context.Context, u *url.URL) (*sql.DB, error) {
+	for k, _ := range u.Query() {
+		// Only permit parameters that do not conflict with other behavior.
+		if k == "user" || k == "password" || k == "dbname" || k == "host" || k == "port" {
+			return nil, fmt.Errorf("postgres: openPostgresURL: extra parameter %s not allowed", k)
+		}
+	}
+
+	db, err := openWithUrl(u)
+	return db, err
+}
+
+// OpenWithUrl temporary Open DB method, for smooth migration to new URL based design.
+func openWithUrl(url *url.URL) (*sql.DB, error) {
+	return sql.Open(Scheme, url.String())
```

Comment:
This should use [`ocsql.Wrap`](https://godoc.org/contrib.go.opencensus.io/integrations/ocsql#Wrap) to add the automatic OpenCensus instrumentation.